myapp/segmentsconfigapp/views.py

```diff
@@ -0,0 +1,85 @@
+import re
+import urlparse
+
+
+from django.shortcuts import render
+from django.core.validators import URLValidator
+from django.core.exceptions import ValidationError
+
+from rest_framework import viewsets
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import AllowAny
+from rest_framework import renderers
+from rest_framework import pagination
+from rest_framework import viewsets
+from rest_framework import request
+from rest_framework.decorators import detail_route, list_route
+from rest_framework.response import Response
+from rest_framework.pagination import PageNumberPagination
+
+from segmentsconfigapp.models import SegmentsConfig
+from segmentsconfigapp.serializers import SegmentsConfigSerializer
+from segmentsconfigapp.parser import ParserSegmentConfig
+
+class SegmentsConfigSetPagination(PageNumberPagination):
+    page_size = 5
+    page_size_query_param = 'page_size'
+    max_page_size = 1000
+
+@permission_classes((AllowAny, ))
+class SegmentsConfigViewSet(viewsets.ModelViewSet):
```

Comment:
If you use a [generic ListCreateAPIView](http://www.django-rest-framework.org/api-guide/generic-views/#listcreateapiview) you wont need to redefine the methods on post or create a new paginator